File: app/src/main/java/com/testapp/testapp/presenter/utils/RequestParametersHolder.java
package com.testapp.testapp.presenter.utils;

import java.util.HashMap;
import java.util.Map;

import static com.testapp.testapp.Constants.Network.*;

/**
 * Created by troll on 05.11.2017.
 */

public class RequestParametersHolder {

    private Map<String, String> requestParameters;

    RequestParametersHolder(){
        requestParameters = new HashMap<>();
        requestParameters.put(DATE_VERSION_FIELD, DateFormatter.getCurrentDateInApiVersionFormat());
        requestParameters.put(CLIENT_ID_FIELD, CLIENT_ID_VALUE);
        requestParameters.put(CLIENT_SECRET_FIELD, CLIENT_SECRET_VALUE);
    }

    private static class Holder{
        static final RequestParametersHolder INSTANCE = new RequestParametersHolder();
    }

    public static RequestParametersHolder getInstance(){
        return Holder.INSTANCE;
    }

    public Map<String, String> getCommonRequestParams(){
        return requestParameters;
    }

    public Map<String, String> getSearchVenuesParamsWithLocation(double longitude, double latitude, String query){
        Map<String, String> searchRequestParams = new HashMap<>();
        searchRequestParams.putAll(requestParameters);
        searchRequestParams.put(LONGITUDE_AND_LATITUDE_FIELD, String.valueOf(latitude) + "," + String.valueOf(longitude));
        searchRequestParams.put(SEARCH_QUERY_FIELD, query);
        searchRequestParams.put(API_INTENT_FIELD, "checkin");
        return searchRequestParams;
    }

    public Map<String, String> getSearchVenuesParamsWithoutLocation(String query){
        Map<String, String> searchRequestParams = new HashMap<>();
        searchRequestParams.putAll(requestParameters);
        searchRequestParams.put(SEARCH_QUERY_FIELD, query);
        searchRequestParams.put(API_INTENT_FIELD, "global");
        return searchRequestParams;
    }

    public Map<String, String> getVenuePhotosParams(){
        Map<String, String> venuePhotosRequestParams = new HashMap<>();
        venuePhotosRequestParams.putAll(requestParameters);
        venuePhotosRequestParams.put(LIMIT_FIELD, String.valueOf(3));
        return venuePhotosRequestParams;
    }
}

File: app/src/main/java/com/testapp/testapp/kotlin/presenter/utils/RequestParametersHolder.kt
package com.testapp.testapp.kotlin.presenter.utils

import com.testapp.testapp.kotlin.Network

class RequestParametersHolder private constructor(){

    private object Holder {
        val INSTANCE = RequestParametersHolder()
    }

    companion object {
        val instance: RequestParametersHolder by lazy { Holder.INSTANCE }
    }

    val requestParameters: MutableMap<String, String> = HashMap()

    init {
        requestParameters[Network.DATE_VERSION_FIELD] = DateFormatter.getCurrentDateInApiVersionFormat()
        requestParameters[Network.CLIENT_ID_FIELD] = Network.CLIENT_ID_VALUE
        requestParameters[Network.CLIENT_SECRET_FIELD] = Network.CLIENT_SECRET_VALUE
    }

    fun getSearchVenuesParamsWithLocation(longitude: Double, latitude: Double, query: String): MutableMap<String, String> {
        val searchRequestParams: MutableMap<String, String> = HashMap()
        searchRequestParams.putAll(requestParameters)
        searchRequestParams[Network.LONGITUDE_AND_LATITUDE_FIELD] = "$latitude,$longitude"
        searchRequestParams[Network.SEARCH_QUERY_FIELD] = query
        searchRequestParams[Network.API_INTENT_FIELD] = "checkin"
        return searchRequestParams
    }

    fun getSearchVenuesParamsWithoutLocation(query: String) : MutableMap<String, String> {
        val searchRequestParams: MutableMap<String, String> = HashMap()
        searchRequestParams.putAll(requestParameters)
        searchRequestParams[Network.SEARCH_QUERY_FIELD] = query
        searchRequestParams[Network.API_INTENT_FIELD] = "global"
        return searchRequestParams
    }

    fun getVenuePhotosParams(): MutableMap<String, String> {
        val venuePhotosRequestParams: MutableMap<String, String> = HashMap()
        venuePhotosRequestParams.putAll(requestParameters)
        venuePhotosRequestParams[Network.LIMIT_FIELD] = "3"
        return venuePhotosRequestParams
    }
}